src/lib/hasher/solana.ts

```diff
@@ -0,0 +1,415 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { createHash } from 'crypto'
+import { PublicKey } from '@solana/web3.js'
+import BN from 'bn.js'
+import { type Schema, serialize } from 'borsh'
+import type { CCIPMessage, CCIPVersion } from '../types.ts'
+import type { LeafHasher } from './common.ts'
+
+// Message types from https://github.com/smartcontractkit/chainlink-ccip/blob/main/chains/solana/contracts/programs/ccip-offramp/src/messages.rs
+/**
+ * RampMessageHeader
+ * - message_id: fixed 32 bytes array.
+ * - source_chain_selector, dest_chain_selector, sequence_number, nonce: u64 values.
+ */
+class RampMessageHeader {
+  message_id: Uint8Array
+  source_chain_selector: BN
+  dest_chain_selector: BN
+  sequence_number: BN
+  nonce: BN
+
+  constructor(fields: {
+    message_id: Uint8Array
+    source_chain_selector: BN
+    dest_chain_selector: BN
+    sequence_number: BN
+    nonce: BN
+  }) {
+    this.message_id = fields.message_id
+    this.source_chain_selector = fields.source_chain_selector
+    this.dest_chain_selector = fields.dest_chain_selector
+    this.sequence_number = fields.sequence_number
+    this.nonce = fields.nonce
+  }
+}
+
+/**
+ * Any2SVMRampExtraArgs
+ * - compute_units: u32
+ * - is_writable_bitmap: u64
+ */
+class Any2SVMRampExtraArgs {
+  compute_units: number
+  is_writable_bitmap: BN
+
+  constructor(fields: { compute_units: number; is_writable_bitmap: BN }) {
+    this.compute_units = fields.compute_units
+    this.is_writable_bitmap = fields.is_writable_bitmap
+  }
+}
+
+/**
+ * CrossChainAmount
+ * - le_bytes: fixed 32 bytes array.
+ */
+class CrossChainAmount {
```

Comment:
This will already come as bigint for you from atlas; no need for a class only for this struct?